### ISSUE TITLE
Slotted runtime fails on OPTIONAL MATCH and double aggregation

### ISSUE THREAD

**Issue by szarnyasg** [neo4j/neo4j#10910]
- Neo4j version: 3.4.0.alpha06 enterprise
- Operating system: Ubuntu 16.04
- API/Driver: any

I was experimenting with some queries on the LDBC Social Network Benchmark's data set and ran into an error when running a query. I managed to simplify it to this:

```
MATCH (person:Person)
OPTIONAL MATCH (person)<-[:HAS_CREATOR]-(message:Message)-[:REPLY_OF*0..]->(post:Post)
WITH person, count(message) AS messageCount
RETURN messageCount, count(person) AS personCount
```

The error message is:

```
Ouch, no suitable slot for key post = NO_VALUE
Slots: SlotConfiguration(longs=1, refs=0, slots=Map(person -> LongSlot(0,false,Node)))
```

Prefixing the query with `CYPHER runtime=interpreted` makes the error go away, so this must be an issue with the compiled runtime.

Reproducing the error needs some data. The scale factor 0.1 data set of the LDBC is sufficient for this goal:
* [zipped `graph.db` (Dropbox)](https://www.dropbox.com/s/z8l7o0yellsattr/graph.db.zip?dl=0)
* [CSV files with import instructions (GitHub)](https://github.com/szarnyasg/ldbc-sf01)

The debug.log is available at https://gist.github.com/szarnyasg/7c2bf5c026d3295116004946b193dbda.

I am cc-ing @alexaverbuch who helped me debug the cause of this issue.

**Comment by szarnyasg** [comment]
I also got a similar error with the following query:

```
MATCH
  (:TagClass {name: 'MusicalArtist'})<-[:HAS_TYPE]-(:Tag)<-[:HAS_TAG]-
  (forum1:Forum)-[:HAS_MEMBER]->(stranger:Person)
WITH DISTINCT stranger
MATCH
  (:TagClass {name: 'Writer'})<-[:HAS_TYPE]-(:Tag)<-[:HAS_TAG]-
  (forum2:Forum)-[:HAS_MEMBER]->(stranger)
WITH DISTINCT stranger
MATCH
  (person:Person)<-[:HAS_CREATOR]-(:Message)-[:REPLY_OF]-
  (:Message)-[:HAS_CREATOR]->(stranger)
WHERE person.birthday > 19881231
  AND person <> stranger
  AND NOT (person)-[:KNOWS]-(stranger)
WITH person, stranger
OPTIONAL MATCH
  (person)<-[:HAS_CREATOR]-(comment1:Comment)-[:REPLY_OF]->(:Message)-[:HAS_CREATOR]->(stranger)
OPTIONAL MATCH
  (stranger)<-[:HAS_CREATOR]-(comment2:Comment)-[:REPLY_OF]->(:Message)-[:HAS_CREATOR]->(person)
WITH
  person,
  count(stranger) AS strangersCount,
  count(comment1) AS comment1Count,
  count(comment2) AS comment2Count
RETURN
  person.id,
  strangersCount,
  comment1Count + comment2Count AS interactionCount
ORDER BY
  interactionCount DESC,
  person.id ASC
LIMIT 100
```

Added the log to https://gist.github.com/szarnyasg/7c2bf5c026d3295116004946b193dbda#file-debug-2-log.

**Comment by Lojjs** [comment]
Thanks for your detailed bug report @szarnyasg We in the Cypher team will have a look at this and come back to you.

**Comment by SaschaPeukert** [comment]
Thank you for reporting. We managed to reproduce this on 3.4.0.alpha06 but not in 3.4.0.alpha08. Can you confirm that it is closed for you in 3.4.0.alpha08? Also, how long do you expect the second query to take? It's not crashed for me yet but isn't finished either

**Comment by szarnyasg** [comment]
I grabbed alpha08: the first query runs fine. The second indeed takes a lot of time (but it does not crash as it used to), as it does a big antijoin operation followed by a double optional match.

Anyways, I don't think that's a problem: I changed that query a lot since then (see its [commit history](https://github.com/ldbc/ldbc_snb_implementations/commits/master/cypher/queries-opencypher/bi-19.cypher)). It turned out I misinterpreted the LDBC spec in one place and we also changed the semantics of the query in another. The new query runs a few seconds using alpha08.

So in conclusion, I believe this issue is fixed now.
